src/providers/Provider.py

```diff
@@ -1,12 +1,55 @@
 from abc import ABC, abstractmethod
 from typing import List, Tuple
+import multiprocessing
+
+from src.Product import Product
+from src.loggers import Logger
+
+
+def flatten(list_of_lists: List[List[object]]) -> List[object]:
```

Comment:
I would not recommend having `NumPy` as a dependency of this project, if `flatten` is the only functionality we need from it, it would be killing a bird with an F-35

Reply:
Despite the fact that a bird will never survive an encounter with an F-35, this discussion is irrelevant as this was changed and flatten is no longer used